### ISSUE TITLE
RefineMesh - std::bad_alloc 

### ISSUE THREAD

**Issue by BigAloz** [cdcseacave/openMVS#276]
### Specifications like the version of the project, operating system, and hardware
I have installed openMVS on the Amazon Web Service.
I have installed on a Ubuntu machine using the instructions from this website.
My current machine has 64GB ram and plenty of cores (I can check from here but would be around 20 cores). I am not using the GPU.

I can upscale to more cores,RAM and a GPU compute if that is what the problem is related to but would have thought I have plenty of computing memory for this task.

### Steps to reproduce the problem
It was working fine until my current data set.
1. 28 images of around 2.2 MB in size.
2. this is quite a relatively small data set

I get a std::bad_alloc in RefineMesh which I assume means it has run out of memory.
- I tried lower the max-threads setting as I assume less threads means less memory usage
- I also tried --reduce-memory options

I can't find  a solution and was wondering if there other settings i need to try in the pipeline processing pipeline.

I am new to Linux so haven't been able to use gdb to try and identify the line that is causing the problem.

Any advice will be appreciated.

Regards,

Alan Buchanan

**Comment by cdcseacave** [comment]
This does not seem to be a memory issue judging by the number of images. Is the SfM reconstruction looking good? You can validate it visually using the Viewer included in _OpenMVS_ and the `scene.mvs you try to reconstruct.

**Comment by BigAloz** [comment]
​
 mvs.zip
<https://drive.google.com/file/d/1cZYS_LaXzBzgauVr9ymWgAwkZxSVMwWV/view?usp=drive_web>
​
The dataset is fine (reprojection error of around 0.6 pixels). I have been
using this for testing for quite some time.

I generated the scene.mvs from openMVG.

I haven't worked out how to debug and step through the program to isolate
the line that is causing this issue.

I have stuck the dataset on the cloud if you have time to check itv out.

Regards,

Alan

**Comment by cdcseacave** [comment]
The reconstruction looks good indeed, the problem is that you geo-reference the scene during SfM; due to reasons explained so many times in previous issues, you should do that only after MVS.

**Comment by BigAloz** [comment]
Ok thank you.

So related to you using floats in the algorithm.

I'll try it using a different coordinates system

By the way is there a limit on the amount of images that can be processed ?
With processing on the cloud I have access to a large amount of cores and
RAM.

I have a dataset with 1500 images which I would like run through this
pipeline

**Comment by cdcseacave** [comment]
No limit, give it a try, depends also a lot on the quality settings you use. You can also use OpenMVG clustering algorithm to split the scene in multiple smaller parts and run OpenMVS on each of them.

**Comment by cdcseacave** [comment]
Closing as stale. The OpenMVS codebase has undergone extensive refactoring and modernisation on the \develop\ branch since this issue was opened. If you believe this is still relevant, please test against the latest \develop\ branch and open a new issue with reproduction steps.
